src/app/context/SettingsContext.tsx

```diff
@@ -47,6 +48,11 @@ export const SettingsProvider = ({
       });
   }, []);
 
+  // set locale from settings on mount
+  useEffect(() => {
+    i18n.changeLanguage(settings.locale);
+  }, [settings]);
+
```

Comment:
Now this hook is called everytime the "settings" change.
I would use settings.locale instead.

```suggestion
 // update i18n on every locale change
  useEffect(() => {
    i18n.changeLanguage(settings.locale);
  }, [settings.locale]);
```